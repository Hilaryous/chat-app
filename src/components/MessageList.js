// @flow
import React from 'react'
import map from 'ramda/src/map'
import AppContext from '../context'
import Message from './Message'
import type { MessageType } from '../types/Message'

type Props = {
  list: Array<MessageType>
}

const MessageList = ({ list }: Props) => (
  <div>
    { map(item => <Message key={item.content} message={item} />, list)}
  </div>
)

export default () => (
  <AppContext.Consumer>
    {({ messages }) => <MessageList list={messages} />}
  </AppContext.Consumer>
)

// --------------------------
