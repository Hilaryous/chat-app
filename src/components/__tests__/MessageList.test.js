import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'
import MessageList from '../MessageList'
import AppContext from '../../context'

describe('MessageList', () => {
  it('shows the correct snapshot tree for the MessageList component', () => {
    const tree = renderer.create(
      <StaticRouter location="someLocation" context={{}}>

        <AppContext.Provider>
          <MessageList />
        </AppContext.Provider>
      </StaticRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
