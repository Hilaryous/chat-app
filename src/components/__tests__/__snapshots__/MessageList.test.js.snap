// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList shows the correct snapshot tree for the MessageList component 1`] = `<div />`;
